assets/source/setup-guide/app/components/Account/Connection.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import apiFetch from '@wordpress/api-fetch';
+import { Spinner } from '@woocommerce/components';
+import { getNewPath } from '@woocommerce/navigation';
+import {
+	Button,
+	CardBody,
+	Flex,
+	FlexItem,
+	FlexBlock,
+	Modal,
+	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { useCreateNotice } from '../../helpers/effects';
+
+const PinterestLogo = () => {
+	return (
+		<>
+			<img
+				src={
+					wcSettings.pinterest_for_woocommerce.pluginUrl +
+					'/assets/images/pinterest-logo.svg'
+				}
+				alt=""
+			/>
+		</>
+	);
+};
+
+const AccountConnection = ( { isConnected, setIsConnected, accountData } ) => {
+	const createNotice = useCreateNotice();
+
+	const [ isConfirmationModalOpen, setIsConfirmationModalOpen ] = useState(
+		false
+	);
+
+	const openConfirmationModal = () => {
+		setIsConfirmationModalOpen( true );
+	};
+
+	const closeConfirmationModal = () => {
+		setIsConfirmationModalOpen( false );
+	};
+
+	const renderConfirmationModal = () => {
+		return (
+			<Modal
+				title={
+					<>{ __( 'Are you sure?', 'pinterest-for-woocommerce' ) }</>
+				}
+				onRequestClose={ closeConfirmationModal }
+				className="woocommerce-setup-guide__step-modal"
+			>
+				<div className="woocommerce-setup-guide__step-modal__wrapper">
+					<p>
+						{ __(
+							'Are you sure you want to disconnect this account?',
+							'pinterest-for-woocommerce'
+						) }
+					</p>
+					<div className="woocommerce-setup-guide__step-modal__buttons">
+						<Button
+							isDestructive
+							isSecondary
+							onClick={ handleDisconnectAccount }
+						>
+							{ __(
+								"Yes, I'm sure",
+								'pinterest-for-woocommerce'
+							) }
+						</Button>
+						<Button isTertiary onClick={ closeConfirmationModal }>
+							{ __( 'Cancel', 'pinterest-for-woocommerce' ) }
+						</Button>
+					</div>
+				</div>
+			</Modal>
+		);
+	};
+
+	const handleDisconnectAccount = async () => {
+		closeConfirmationModal();
+
+		const result = await apiFetch( {
+			path:
+				wcSettings.pinterest_for_woocommerce.apiRoute +
+				'/auth_disconnect',
+			method: 'POST',
+		} );
```

Comment:
Missing try/catch block and error handling (e.g. lost network connection) for API call.